evmrpc/tx.go

```diff
@@ -53,6 +53,44 @@ func (t *TransactionAPI) GetTransactionReceipt(ctx context.Context, hash common.
 		}
 		return nil, err
 	}
+	// Fill in the receipt if the transaction has failed and used 0 gas
+	if receipt.TxType == 0 && receipt.GasUsed == 0 {
+		// Get the block
+		height := int64(receipt.BlockNumber)
+		block, err := blockByNumberWithRetry(ctx, t.tmClient, &height, 1)
+		if err != nil {
+			return nil, err
+		}
+
+		// Find the transaction in the block
+		for _, tx := range block.Block.Txs {
+			etx := getEthTxForTxBz(tx, t.txConfig.TxDecoder())
+			if etx != nil && etx.Hash() == hash {
+				// Get the signer
+				signer := ethtypes.MakeSigner(
+					types.DefaultChainConfig().EthereumConfig(t.keeper.ChainID(sdkctx)),
+					big.NewInt(height),
+					uint64(block.Block.Time.Unix()),
+				)
+				from, _ := ethtypes.Sender(signer, etx)
+
+				// Update receipt with correct information
+				receipt.From = from.Hex()
+				if etx.To() != nil {
+					receipt.To = etx.To().Hex()
+					receipt.ContractAddress = ""
+				} else {
+					receipt.To = ""
+					// For contract creation transactions, calculate the contract address
+					receipt.ContractAddress = crypto.CreateAddress(from, etx.Nonce()).Hex()
+				}
+				receipt.TxType = uint32(etx.Type())
+				receipt.Status = uint32(ethtypes.ReceiptStatusFailed)
+				receipt.GasUsed = 0
```

Comment:
i think you can also fill other fields like input, value, etc. from `etc`

Reply:
it looks receipt doesn't have these fields: https://github.com/sei-protocol/sei-chain/blob/main/x/evm/types/receipt.pb.go#L102

i checked some successfully tx receipts, they have logs but it doesn't have input/value either.